test/integration/confirmations/signatures/personalSign.test.tsx

```diff
@@ -0,0 +1,259 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+import { ApprovalType } from '@metamask/controller-utils';
+import mockMetaMaskState from '../../data/integration-init-state.json';
+import { integrationTestRender } from '../../../lib/render-helpers';
+import { shortenAddress } from '../../../../ui/helpers/utils/util';
+import * as backgroundConnection from '../../../../ui/store/background-connection';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsEventLocation,
+} from '../../../../shared/constants/metametrics';
+
+jest.mock('../../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+}));
+
+const mockedBakcgroundConnection = backgroundConnection as jest.Mocked<
+  typeof backgroundConnection
+>;
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+describe('PersonalSign', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  afterAll(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  it('displays the header account modal with correct data', async () => {
+    const account =
+      mockMetaMaskState.internalAccounts.accounts[
+        mockMetaMaskState.internalAccounts
+          .selectedAccount as 'cf8dace4-9439-4bd4-b3a8-88c821c8fcb3'
```

Comment:
Why are we casting this to a specific string?

Reply:
Because typescript is inferring the types from the mockMetaMaskState.internalAccounts.accounts object keys and not setting it up as just a string type... I wanted to avoid typing the mockMetaMaskState and as it was a test case I didn't spend much time over it.
Maybe we can update that casting to `as keyof typeof mockMetaMaskState.internalAccounts.accounts`? The only other solution that I can think off is typing `mockMetaMaskState`... Do you have any ideas?